tests/net/ieee802154/l2/src/ieee802154_test.c

```diff
@@ -177,6 +177,44 @@ static int tear_down_short_addr(struct net_if *iface, struct ieee802154_context
 	return ret;
 }
 
+static bool set_up_security(uint8_t security_level)
+{
+	struct ieee802154_security_params params;
+
+	if (security_level == IEEE802154_SECURITY_LEVEL_NONE) {
+		return true;
+	}
+
+	params = (struct ieee802154_security_params){
+		.key = {0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb,
+			0xcc, 0xcd, 0xce, 0xcf},
+		.key_len = 16U,
+		.key_mode = IEEE802154_KEY_ID_MODE_IMPLICIT,
+		.level = security_level,
+	};
+
+	if (net_mgmt(NET_REQUEST_IEEE802154_SET_SECURITY_SETTINGS, iface, &params,
+		     sizeof(struct ieee802154_security_params))) {
+		NET_ERR("*** Failed to set security settings\n");
+		return false;
+	}
+
+	return true;
+}
+
+static void tear_down_security(void)
+{
+	struct ieee802154_context *ctx = net_if_l2_data(iface);
+
+	if (ctx->sec_ctx.level == IEEE802154_SECURITY_LEVEL_NONE) {
+		return;
+	}
+
+	cipher_free_session(ctx->sec_ctx.enc.device, &ctx->sec_ctx.enc);
+	cipher_free_session(ctx->sec_ctx.dec.device, &ctx->sec_ctx.dec);
+	ctx->sec_ctx.level = IEEE802154_SECURITY_LEVEL_NONE;
```

Comment:
Did you know why it is necessary clean-up using cipher* functions instead make a request passing security_level as IEEE802154_SECURITY_LEVEL_NONE ? This seems to be a little hack to me.

Reply:
This is a very good question. The fact that the security context is not cleaned up inside the mgmt-stack before starting a new one is even a resource leak. Fixed.